src/transformers/trainer.py

```diff
@@ -1697,6 +1697,8 @@ def _inner_training_loop(
         use_accelerator_prepare = True if model is self.model else False
 
         if delay_optimizer_creation:
+            if use_accelerator_prepare:
+                self.model = self.accelerator.prepare(self.model)
```

Comment:
are we going to prepare twice with `model = self.accelerator.prepare(self.model)` and `model, self.optimizer = self.accelerator.prepare(self.model, self.optimizer)` ? (+ this one)

Reply:
In accelerate, the second call to prepare won't do anything to the model as it was already prepared and will only prepare the rest of the things.

Reply:
Alright.